[documentation]: Restructure docs and add docs for new features (#16)
txtdirect/website: content/docs/config/_index/content.md

+++
fragment = "content"
title = "Configuration"
weight = 100

[sidebar]
  enable = true
+++

TXTDirect uses [Caddy's](https://caddyserver.com) config controller so all the config files should follow the [Caddyfile syntax](https://caddyserver.com/v1/tutorial/caddyfile) rules.

> TXTDirect's reads its config from the `txtdirect` scope in the Caddyfile. So for example, using the following config you can tell TXTDirect to enable the `host` redirect type on `example.com` host and write the logs to `stdout`:

Reviewer: ```suggestion
TXTDirect's reads its config from the `txtdirect` scope in the Caddyfile. For example, using the following config you can tell TXTDirect to enable the `host` redirect type on the `example.com` host and write the logs to `stdout`:
```

```
example.com:8080 {
  errors stdout
  txtdirect {
    enable host
    logfile stdout
  }
}
```

Some of TXTDirect's features like Prometheus metrics or QR need to be configured separately.
These features need to be configured inside a separate scope like the example below:

```
example.com:8080 {
  txtdirect {
    enable host
    prometheus {
      address 192.168.0.1
    }
  }
}
```

> These features are documented in separate sections that you can find using the menu on left side of the page.

Reviewer: Binding content to the layout is problematic. Maybe we can reword this to work without taking into account the layout?

Author: Would it sound better if we only mentioned the documentation navigation menu and not its position on the page? Can't really think of a better way to point to the menu.

Reviewer: That would be a start. Or we could link to the docs in general as the overview page should lead them there as well.
Something along the lines of.
These features are documented in more detail in separate sections within our [docs].

## CLI Flags

> Use the `-conf` flag to pass the config file to TXTDirect like this example:

Reviewer: ```suggestion
Use the `-conf` flag to pass a config file to TXTDirect.
```

}
```

These features are documented in more detail in separate sections within our

> [documentation](/docs) under the configuration category.

Reviewer: Why not link to the specific section?
